musicbot/bot.py

```diff
@@ -4742,49 +4742,51 @@ async def cmd_summon(
         Call the bot to the summoner's voice channel.
         """
 
-        # @TheerapakG: Maybe summon should have async lock?
-
-        if not author.voice or not author.voice.channel:
-            raise exceptions.CommandError(
-                self.str.get(
-                    "cmd-summon-novc",
-                    "You are not connected to voice. Try joining a voice channel!",
+        lock_key = f"summon:{guild.id}"
+        log.info("Attempting to acquire lock for %s", lock_key)
+        async with self.aiolocks[lock_key]:
+            log.info("Attempting to acquire lock for %s", lock_key)
```

Comment:
The log lines here should probably demoted to debug or another appropriate verbosity level.
I would also suggest to change the log text to `Waiting for summon lock...` and `Summon lock acquired.`  since that is a bit more clear on what is going on.  

If you're feeling extra fancy, you can use the state of the lock `self.aiolocks[lock_key].locked()` to determine if we should even log that we waited at all.   Not a requirement.  We'll be doing UI stuff later on down the line.

Reply:
Do you want to keep the `%s, lock_key`?

Reply:
You can keep the lock key, or just include the guild name.  

In regards to using `locked()` to control output, you'll want to check and store the locked state before you enter the `async with` block.   After you enter the locked block, the call to `locked()` will always return `True` since the block of code has acquired the lock in order to execute.